pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -106,6 +97,42 @@ func (r *ReconcileNSTemplateSet) Reconcile(request reconcile.Request) (reconcile
 		reqLogger.Error(err, "failed to get NSTemplateSet")
 		return reconcile.Result{}, err
 	}
+	if util.IsBeingDeleted(nsTmplSet) {
+		// if the NSTemplateSet has no finalizer, then we don't have anything to do
+		if len(nsTmplSet.Finalizers) == 0 {
```

Comment:
you could use github.com/redhat-cop/operator-utils for manipulating with finalizers - see how it is done for UserAccount: https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller.go#L292-L302 or MUR: https://github.com/codeready-toolchain/host-operator/blob/master/pkg/controller/masteruserrecord/masteruserrecord_controller.go#L118-L152
The same repo can be used also for removal of the finalizer - use `RemoveFinalizer` function

Reply:
We should not check just if there is any finalizer, but if there is our specific finalizer `toolchainv1alpha1.FinalizerName`

Reply:
hmm, that's already done: https://github.com/codeready-toolchain/member-operator/pull/121/commits/e3dbfff3555c5e08d381398fb9a1994e59faf9d3

Reply:
I think this is what @MatousJobanek means: https://github.com/codeready-toolchain/member-operator/pull/121#discussion_r374371849

Reply:
ok, got it now! thanks for the clarification

Reply:
done in b1f9439